dagger-common/src/test/java/com/gotocompany/dagger/common/serde/typehandler/TypeHandlerFactoryTest.java

```diff
@@ -12,19 +16,47 @@
 import com.gotocompany.dagger.consumer.TestRepeatedEnumMessage;
 import org.junit.Before;
 import org.junit.Test;
-
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.mockito.Mockito.when;
+
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
 
+@PrepareForTest(Descriptors.FieldDescriptor.class)
+@RunWith(PowerMockRunner.class)
 public class TypeHandlerFactoryTest {
     @Before
     public void setup() {
         TypeHandlerFactory.clearTypeHandlerMap();
     }
 
+    @Test
+    public void shouldReturnTheSameHandlerObjectWhenBothFieldDescriptorFullNameAndFieldDescriptorHashCodeIsSame() {
+        Descriptors.FieldDescriptor mapFieldDescriptor = TestBookingLogMessage.getDescriptor().findFieldByName("metadata");
+        TypeHandler typeHandler1 = TypeHandlerFactory.getTypeHandler(mapFieldDescriptor);
+        TypeHandler typeHandler2 = TypeHandlerFactory.getTypeHandler(mapFieldDescriptor);
+        assertEquals(typeHandler1, typeHandler2);
+    }
+
+    @Test
+    public void shouldReturnDifferentCopiesOfHandlerObjectWhenFieldDescriptorFullNameIsSameButHashCodeIsDifferent() {
+        Descriptors.FieldDescriptor mapFieldDescriptor1 = TestBookingLogMessage.getDescriptor().findFieldByName("metadata");
+        Descriptors.FieldDescriptor mapFieldDescriptor2 = PowerMockito.mock(Descriptors.FieldDescriptor.class);
```

Comment:
if you again create this, how the hashcode will be same?

Reply:
Changed to use protobuf libraries